code/modules/power/generator.dm

```diff
@@ -113,12 +125,29 @@
 		circ2.network2.update = 1
 
 	//Exceeding maximum power leads to some power loss
-	if(effective_gen > max_power && prob(5))
-		var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
-		s.set_up(2, 1, src)
-		s.start()
-		stored_energy *= 0.5
-
+	//exceeding max power is supposed to be bad, have there be a small chance of some visual and audio effects to stress this
+	//remember to lubricate your engines kiddos
+	if(!lubricated)
+		if(effective_gen > max_power && prob(5))
+			var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
```

Comment:
That's overly verbose and inconsistent with the way you do it for smoke a couple lines below. Speaking of which - you have a /effect/system for sparks called `s`, and one for smoke called `SS` - this is gonna be confusing to maintain.